libraries/WiFiClientSecure/src/ssl_client.cpp

```diff
@@ -19,6 +19,9 @@
 #include "ssl_client.h"
 #include "WiFi.h"
 
+#ifndef MBEDTLS_KEY_EXCHANGE__SOME__PSK_ENABLED
```

Comment:
This would need to be prefixed with CONFIG_... But I don't see this key on IDF v3.2 or IDF v3.3.

Reply:
CONFIG_MBEDTLS_PSK_MODES and CONFIG_MBEDTLS_TLS_ENABLED would be better to check for here.

Reply:
This is not quite easy to handle with the CONFIG_* definitions because internally the library will use a lot of things together (the basic config must be set + at least one cipher must be selected) to enable this function. So I choose to use the library definition which really enables the function to not have duplicate logic in there.

Reply:
The problem is the define you have used will never exist, regardless of sdkconfig entries selected. If you switch to the CONFIG_XXX entries I posted above it will at least compile for test cases.

But you are right, there is still the requirement to have at least one key exchange format enabled which will require explicit checking for the various keys used by the code.